code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -50,13 +50,31 @@
 	see_in_dark = 4
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
 	var/playable_spider = FALSE
+	var/datum/action/innate/lay_web/lay_web
+	var/directive = "" //Message passed down to children, to relay the creator's orders
+
+/mob/living/simple_animal/hostile/poison/giant_spider/Initialize()
+	. = ..()
+	lay_web = new
+	lay_web.Grant(src)

```

Comment:
shouldn't you only grant these when there is a mob in the spider?

Reply:
same goes for the rest of them?

i.e do it in login /logout not intialize

Reply:
ah I see, you're using it for the AI in the mob as well

probably not what I would have done (had sub procs to share the behaviour between the UI grants and the mob ai, but that's fien I guess

Reply:
buttons from actions are only processed when there's a client/hud attached anyway so using actions in this way is completely fine.